### ISSUE TITLE
Victoria Metrics crashes when importing data from influx using vmctl

### ISSUE THREAD

**Issue by shweppsie** [VictoriaMetrics/VictoriaMetrics#481]
**Describe the bug**
When I try to import data into VictoraMetrics from influx using vmctl VictoriaMetrics crashes. If I have vmagent in the middle vmagent crashes.

I've had a read of the source code and can see it's dying here: https://github.com/VictoriaMetrics/VictoriaMetrics/blob/21598ac4172fedccf81251ddd56efb490d99e774/app/vmagent/remotewrite/pendingseries.go#L150
I'm happy to help fix the issue but I'm not 100% on what the code is doing at this point. Potentially it's a bug in vmctl but I'd need some pointers to know where to start.

**To Reproduce**
Try to import influx data into VictoriaMetrics. If I restart the import it happens every time.

**Expected behavior**
Data is imported.

**Version**
```
victoria-metrics-20200505-082206-tags-v1.35.2-0-g0c7cddfc
```

```
vmagent-20200507-201403-tags-v1.35.3-0-gefc8e3c5
```

**Used command-line flags**
```
./vmctl influx --influx-addr http://10.6.2.4:8086 --influx-database prometheus2 --vm-addr http://10.6.0.13:8429
```
docker-compose.yml for vmagent
```
version: "2.4"

services:
  vmagent:
    build:
      context: .
    command:
      - "-remoteWrite.url=http://10.6.0.13:8428/api/v1/write"
      - "-remoteWrite.relabelConfig=/etc/victoriametrics/relabel.yml"
    volumes:
      - "/storage/vmagent/:/vmagent-remotewrite-data/"
    ports:
      - "8429:8429"
```
relabel.yml
```
 - regex: prometheus_instance
   action: labeldrop
 - source_labels: [ __name__ ]
   regex: ALERTS
   action: drop
 - source_labels: [ __name__ ]
   regex: ALERTS_FOR_STATE
   action: drop
```
docker-compose.yml for victoriametrics
```
version: "2.4"

services:
  victoriametrics:
    image: victoriametrics/victoria-metrics
    ports:
      - "8428:8428"
    volumes:
      - "/storage/victoriametrics/:/victoria-metrics-data"
```

**Additional context**
```
vmagent_1  | 2020-05-14T09:43:22.137Z	panic	VictoriaMetrics/app/vmagent/remotewrite/pendingseries.go:150	BUG: unexpected number of samples in time series; got 4; want 1
vmagent_1  | 2020-05-14T09:43:22.137Z	error	net/http/server.go:3059	http: panic serving 10.6.0.13:45728: BUG: unexpected number of samples in time series; got 4; want 1
vmagent_1  | goroutine 14426790 [running]:
vmagent_1  | net/http.(*conn).serve.func1(0xc0000f6a00)
vmagent_1  | 	net/http/server.go:1772 +0x139
vmagent_1  | panic(0xa78620, 0xc0004618a0)
vmagent_1  | 	runtime/panic.go:975 +0x3e3
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage(0xb13183, 0x5, 0xc00068e580, 0x3f, 0x4)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:188 +0xa65
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, 0xb13183, 0x5, 0xb37bcf, 0x40, 0xc000a75410, 0x1, 0x1)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:121 +0xd0
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel(...)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:113
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf(...)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:109
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*writeRequest).copyTimeSeries(0xc0006b4008, 0xc001cfc440, 0xc000552800)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/pendingseries.go:150 +0x6fb
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*writeRequest).push(0xc0006b4008, 0xc000552800, 0x3, 0x80)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/pendingseries.go:121 +0x115
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*pendingSeries).Push(0xc0006b4000, 0xc000552800, 0x3, 0x80)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/pendingseries.go:53 +0x63
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*remoteWriteCtx).Push(0xc0001c60a0, 0xc000552800, 0x3, 0x80)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:191 +0x9e
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.Push(0xc0000dc050)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:111 +0x91
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/vmimport.insertRows(0xc0000fc360, 0x3, 0x4, 0x0, 0x0)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/vmimport/request_handler.go:66 +0x6de
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/lib/protoparser/vmimport.ParseStream(0xc000a3d400, 0xb45328, 0x0, 0x0)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/protoparser/vmimport/streamparser.go:38 +0x17b
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/vmimport.InsertHandler.func1(0xc0006ae0c0, 0xc000a5e8c8)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/vmimport/request_handler.go:25 +0x36
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/lib/writeconcurrencylimiter.Do(0xc000a5e9a8, 0x7f6d79ead500, 0x203000)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/writeconcurrencylimiter/concurrencylimiter.go:37 +0x2fc
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/vmimport.InsertHandler(0xc000a3d400, 0xe, 0xb11ff4)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/vmimport/request_handler.go:24 +0x49
vmagent_1  | main.requestHandler(0xbed3a0, 0xc000dd9d70, 0xc000a3d400, 0x1)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/main.go:124 +0x703
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0xc000112420, 0xbed3a0, 0xc000dd9d70, 0xc000a3d400, 0xb45840)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:236 +0x340
vmagent_1  | github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1(0xbed3a0, 0xc000dd9d70, 0xc000a3d400)
vmagent_1  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:159 +0xa7
vmagent_1  | net/http.HandlerFunc.ServeHTTP(0xc0008021c0, 0xbed7a0, 0xc000a3eee0, 0xc000a3d400)
vmagent_1  | 	net/http/server.go:2012 +0x44
vmagent_1  | net/http.serverHandler.ServeHTTP(0xc00049c000, 0xbed7a0, 0xc000a3eee0, 0xc000a3d400)
vmagent_1  | 	net/http/server.go:2807 +0xa3
vmagent_1  | net/http.(*conn).serve(0xc0000f6a00, 0xbeed20, 0xc000a39980)
vmagent_1  | 	net/http/server.go:1895 +0x86c
vmagent_1  | created by net/http.(*Server).Serve
vmagent_1  | 	net/http/server.go:2933 +0x35c
```
vmctl output
```
# ./vmctl influx --influx-addr http://10.6.2.4:8086 --influx-database prometheus2 --vm-addr http://10.6.0.13:8429
InfluxDB import mode
2020/05/14 21:41:08 Exploring scheme for database "prometheus2"
2020/05/14 21:41:08 fetching fields: command: "show field keys"; database: "prometheus2"; retention: "autogen"
2020/05/14 21:41:09 found 2091 fields
2020/05/14 21:41:09 fetching series: command: "show series"; database: "prometheus2"; retention: "autogen"
2020/05/14 21:42:19 found 7481123 series
Found 7481123 timeseries to import. Continue? [Y/n] Y
55 / 7481123 [>___________________________________________________________________________________________________________] 0.00% 0 p/s
```

**Comment by shweppsie** [comment]
Actually, I was wrong. This only happens via vmagent, which makes sense seeing as that's where the code is. I'd like to drop some series on import, and remove an external label I added to support thanos. Is it possible to backfill through vmagent?

**Comment by shweppsie** [comment]
I've solved my problem by modifying vmctl to allow me to selectively drop labels during import, so if this isn't actually a bug feel free to close it off.

**Comment by hagen1778** [comment]
Hi @shweppsie! Could you pls provide an example of labels which you dropped?

**Comment by valyala** [comment]
Confirmed the bug. It triggers when importing multiple samples for the same time series via `/api/v1/import` . The issue will be fixed soon.

**Comment by valyala** [comment]
@shweppsie , could you build `vmagent` from the commit 93267f143f8ca9c89eac90fdf012d32d8ba6d2e0 and try importing multiple samples for the same time series via `/api/v1/import`? See [instructions on how to build vmagent from sources](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vmagent/README.md#how-to-build-from-sources).

**Comment by valyala** [comment]
oops, the commit mentioned above is broken. Could you build and verify `vmagent` from the commit 697b6af10f3f6851b5def76402c10f4509d48c5a instead?

**Comment by shweppsie** [comment]
Awesome! Yeah, I'll give it a go shortly.

**Comment by shweppsie** [comment]
Perfect, that's fixed it. I'll leave it running over the weekend to see if there's any other issues. Thanks!

**Comment by valyala** [comment]
The bugfix has been included in [v1.35.5](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue.
